mypy/checkexpr.py

```diff
@@ -2754,18 +2737,58 @@ def check_overload_call(
             self.chk.fail(message_registry.TOO_MANY_UNION_COMBINATIONS, context)
         return result
 
+    def adjust_generic_callable_params_mapping(
```

Comment:
This doesn't need to be deduplicated anymore because it's only called once

Reply:
Honestly, I'd rather keep this separated: that block is rather long and not absolutely trivial, it looks better as a separate method to my eyes. However, there was another problem not detected by linter for some reason (uhm, sorry, because ARG001 is not enabled?.. Why?..), pushed to remove arguments that are no longer necessary.